edgelet/iotedge/src/lib.rs

```diff
@@ -3,6 +3,7 @@
 #![deny(rust_2018_idioms, warnings)]
 #![deny(clippy::all, clippy::pedantic)]
 #![allow(
+    clippy::box_default,
```

Comment:
Was this somehow required when you moved the build containers to Ubuntu 22.04?

Reply:
Nope it isn't required. Removing the rule...